src/vehicle_use.cpp

```diff
@@ -1049,6 +1053,27 @@ void vehicle::reload_seeds( const tripoint &pos )
     }
 }
 
+static void use_towel( const tripoint &pos )
+{
+    ( void )pos; // TODO soil the towel
+    player &p = g->u;
+    bool slime = p.has_effect( effect_slimed );
+    bool boom = p.has_effect( effect_boomered );
+    bool glow = p.has_effect( effect_glowing );
+    int mult = slime + boom + glow; // cleaning off more than one at once makes it take longer
+    p.remove_effect( effect_slimed ); // able to clean off all at once
+    p.remove_effect( effect_boomered );
+    p.remove_effect( effect_glowing );
+    if( abs( p.has_morale( MORALE_WET ) ) ) {
+        p.rem_morale( MORALE_WET );
+        p.body_wetness.fill( 0 );
+        p.add_msg_if_player( _( "You use the towel to dry off." ) );
+        mult += 1;
```

Comment:
This is partially a duplicate of `iuse::towel` Could you perhaps move the duplicated parts to a common function, then call that?

Reply:
Where should i put such a function?

Reply:
iuse.h is included in this file, so I'd personally put it under iuse::towel.

Reply:
I don't get it: there's already iuse::towel, but it can't be used since it's a method of an item, while a car part is not an item.

Reply:
No, what I was saying is you make a new function `towel_common` or whatever, and put that under `iuse::towel` in `iuse.cpp`. Then you'd call that function here,